src/lib/comms/sol-coap.c

```diff
@@ -784,6 +788,23 @@ on_can_write(void *data, struct sol_socket *s)
         return false;
     }
 
+    /* According to RFC 7641, section 4.5, "since RESET messages are
+     * transmitted unreliably, the client must be prepared in case the
+     * these are not received by the server. Thus, a server can always
+     * pretend that a RESET message rejecting a non-confirmable
+     * notification was lost. If a server does this, it could
+     * accelerate cancellation by sending the following notifications
+     * to that client in confirmable messages".
+     *
+     * By taking this shortcut we reduce memory usage A LOT and are
+     * able to run on very small devices with no memory issues.
+     */
+    if (type == SOL_COAP_TYPE_ACK || type == SOL_COAP_TYPE_RESET
```

Comment:
Sorry for only thinking about this now, the correct check would be `!outgoing->timeout`, i.e. for everything that I don't need to retransmit and was just sent successfully.